packages/core/src/browser/widget-decoration.ts

```diff
@@ -330,6 +330,10 @@ export namespace WidgetDecoration {
          * An array of ranges to highlight the caption.
          */
         readonly highlight?: CaptionHighlight;
+        /**
+         * A notification badge for widgets.
+         */
+        readonly badge?: string;
```

Comment:
how is it related to widget decoration? Should not we have a separate type for widget-tab-decoration?

Reply:
@akosyakov How would you suggest I change it? I did it this way because I thought having a badge property in the base widget decoration would allow different widgets to contribute to it directly.

Reply:
I meant i don't see a point in using `WidgetDecoration`, you don't use anything from it except `badge`. But clients which actually make use of `WidgetDecoration`, don't use `badge`. Introduce something like `WidgetTabDecoration` instead.